internal/repository/postgresql/postgresqluser/user.go

```diff
@@ -0,0 +1,159 @@
+package postgresqluser
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	phonenumber "github.com/basliqlabs/qwest-services/pkg/mobile"
+
+	"github.com/basliqlabs/qwest-services/internal/entity/userentity"
+	"github.com/basliqlabs/qwest-services/pkg/errmsg"
+	"github.com/basliqlabs/qwest-services/pkg/richerror"
+)
+
+func (d *DB) DoesUserNameWithPasswordExist(ctx context.Context, username string, password string) (bool, error) {
+	const op = "postgresqluser.DoesUserNameWithPasswordExist"
+	stmt, err := d.db.Conn().PrepareContext(ctx, `SELECT user_id FROM users WHERE username=$1 AND password_hash=$2`)
+	if err != nil {
+		return false, richerror.New(op).
+			WithKind(richerror.KindUnexpected).
+			WithMessage(errmsg.CantScanQueryResult).
+			WithError(err)
+	}
+
+	row := stmt.QueryRowContext(ctx, username, password)
+
+	userId := new(int)
+	err = row.Scan(userId)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return false, richerror.New(op).
```

Comment:
This is not an error
I mean we use the `richerror` to capture server errors and this is not a server error, we simply didn't find the user with the given hash, so I think `false, nil` is a better return value